message-queue-lib/src/cmr/message_queue/queue/sqs.clj

```diff
@@ -132,18 +134,28 @@
                       ;; Tell SQS how long to wait before returning with no data (long polling).
                       (.setWaitTimeSeconds (Integer. (queue-polling-timeout))))]
     (a/thread
-      (try
-        (u/while-let [rec-result (.receiveMessage sqs-client rec-req)]
-          (when-let [msg (first (.getMessages rec-result))]
-            (let [msg-body (.getBody msg)
-                  msg-content (json/decode msg-body true)]
-              (try
-                (handler msg-content)
-                (.deleteMessage sqs-client queue-url (.getReceiptHandle msg))
-                (catch Throwable e
-                  (error e "Message processing failed for message" (pr-str msg) "on queue" queue-name))))))
-        (catch Throwable e
-          (error  e "Async handler for queue" queue-name "completing."))))))
+      (loop []
+        (try
+          (let [rec-result (.receiveMessage sqs-client rec-req)]
+            (when-let [msg (first (.getMessages rec-result))]
+              (let [msg-body (.getBody msg)
+                    msg-content (json/decode msg-body true)]
+                (try
+                  (handler msg-content)
+                  (.deleteMessage sqs-client queue-url (.getReceiptHandle msg))
+                  (catch Throwable e
+                    (error e "Message processing failed for message" (pr-str msg) "on queue" queue-name))))))
+          ;; Catching this so the next catch block won't - this allows us to exit the thread after a test
+          ;; by throwing an ExcpetionInfo object.
+          (catch clojure.lang.ExceptionInfo e
```

Comment:
Couldn't any Clojure library end up throwing something that would get caught here? I'm thinking you might want to create your own exception type and catch it here.